acm/odh-edge/apps/bike-rental-app/bike-rental-inference-generator-cronjob.yaml

```diff
@@ -0,0 +1,65 @@
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: bike-rental-inference-generator
+spec:
+  schedule: "*/5 * * * *"  # This will run the job every hour
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          containers:
+          - name: test-mlflow-container
+            image: quay.io/devguyio/ab-jq
+            securityContext:
+              allowPrivilegeEscalation: false
+              runAsNonRoot: true
+              seccompProfile:
+                type: "RuntimeDefault"
+              capabilities:
+                drop:
+                - ALL
+            command:
+            - /bin/bash
+            - -c
+            - |
+              echo "Test inference REST web service"
+              echo "Data:"
+              cat /workspace/test-data/data.json
+              # Add your expected response here or remove this line if not needed
+              INDEX=$(( RANDOM % 20 ))
+              
+              # Number of requests between 100 and 1000
+              R_COUNT=$(( RANDOM % 901 + 100 ))
+              # Number of concurrent requests between 100 and R_COUNT
+              R_RATE=$(( RANDOM % ($R_COUNT - 99) + 100 ))
+
+              # Randomly selects a record from data.json
+              RANDOM_ROW=$(cat /workspace/test-data/data.json | jq -r ".dataframe_split.data | .[$INDEX]")
+              PAYLOAD="{\"dataframe_split\": {\"columns\":[ \"day\", \"mnth\", \"year\", \"season\",\"holiday\", \"weekday\", \"workingday\", \"weathersit\", \"temp\", \"hum\", \"windspeed\" ], \"data\":[$RANDOM_ROW]}}"
+              echo "Payload: $PAYLOAD"
+              echo "$PAYLOAD" > /tmp/payload.json
+
+
+              # Call the service
+              echo ""
+              echo "Call service:"
+              echo "Number of requests: ${R_COUNT} concurrent requests: ${R_RATE}"
+              # Use ab for load testing with 100 concurrent requests
+              ab -p /tmp/payload.json -T "application/json" -c $R_RATE -n $R_COUNT http://$(MODEL_NAME)-$(MODEL_VERSION):8080/$(TEST_ENDPOINT)
+            env:
+            - name: MODEL_NAME
+              value: "bike-rental-app-model"  # modify this
+            - name: MODEL_VERSION
+              value: "1"  # modify this if needed
+            - name: TEST_ENDPOINT
+              value: "invocations"  # modify this
+            volumeMounts:
+            - name: test-data-volume
+              mountPath: /workspace/test-data
+          volumes:
+          - name: test-data-volume
+            configMap:
+              name: bike-rental-test-data  # This assumes you have a ConfigMap named bike-rental-test-data with the JSON data
```

Comment:
Should kustomize also create this ConfigMap by default, rather than having the assumption in the comment?

Ensuring that this reference is correct with regards to the name of the ConfigMap can also be handled by kustomize `replacements`, similar to how we ensure that a Route refers to the correct Service [here](https://github.com/opendatahub-io/ai-edge/blob/main/acm/odh-edge/apps/bike-rental-app/kustomization.yaml#L39...L48).

Reply:
Done